default/python/AI/AIstate.py

```diff
@@ -182,9 +181,8 @@ def __setstate__(self, state):
             else:
                 try:
                     aggression = state['character'].get_trait(Aggression).key
-                except Exception as e:
-                    print >> sys.stderr, "Could not find the aggression level of the AI, defaulting to typical."
-                    print >> sys.stderr, e
+                except Exception:
+                    warn("Could not find the aggression level of the AI, defaulting to typical.", exc_info=True)
```

Comment:
This should be an error.  `character` and `aggression` are required and expected aspects of the AI model.  If `character` is missing from `state`, or `Aggression` is missing from `character` or `Aggression` has no key, these are all errors.  Furthermore, this is an unexpected `Exception`, so there will be traceback information suggesting it is more important than a warning.